users/management/commands/notify_users_about_activity.py

```diff
@@ -0,0 +1,339 @@
+    ]):
+        user_data = {
+            'user_id': user['id'],
+            'first_name': user['first_name'],
+            'slack_team_id': user['social_profiles__slack_team_id'],
+            'slack_id': user['social_profiles__slack_id'],
+        }
+        penny_chats = []
+        for penny_chat in grouped(user['items'], [
+            'user_chats__penny_chat__id',
+            'user_chats__penny_chat__title',
+            'user_chats__penny_chat__date',
+        ]):
+            penny_chat_data = {
+                'id': penny_chat['user_chats__penny_chat__id'],
+                'title': penny_chat['user_chats__penny_chat__title'],
+                'date': penny_chat['user_chats__penny_chat__date'],
+            }
+            followups = []
+            for followup in penny_chat['items']:
+                if user['id'] == followup['user_chats__penny_chat__follow_ups__user_id']:
+                    # ignore followups that were created by the user
+                    continue
+                followup_data = {
+                    'user_id': followup['user_chats__penny_chat__follow_ups__user_id'],
+                    'first_name': followup['user_chats__penny_chat__follow_ups__user__first_name'],
+                }
+                followups.append(followup_data)
+            if followups:
+                penny_chat_data['followups'] = followups
+                penny_chats.append(penny_chat_data)
+        if penny_chats:
+            user_data['penny_chats'] = penny_chats
+            yield user_data
+
+
+def notify_about_activity(user_data, live_run=False):
+    """Notifies users in slack about recent activity in Penny Chat's they've participated in.
+
+    :param user_data: expected form is
+        {
+            'user_id': ...,
+            'first_name': ...,
+            'slack_team_id': ...,
+            'slack_id': ...,
+            'penny_chats': [{
+               'id': ...,
+               'title': ...,
+               'date': ...,
+               'followups': [{
+                   'user_id': ...,
+                   'first_name': ...,
+                }, ...],
+            }, ...],
+        }
+
+    :return: Nothing. Notifies users as side effect.
+    """
+    chat_lines = []
+    for penny_chat in user_data['penny_chats']:
+        people = {
+            followup['first_name']: followup['user_id']
+            for followup in penny_chat['followups']
+            if followup['user_id'] != user_data['user_id']
+        }
+        if not people:
+            continue
+        people_string = get_people_string(people)
+        date_string = (
+            f'<!date^{int(penny_chat["date"].timestamp())}^{{date_short}}|{penny_chat["date"].strftime("%b %d, %Y")}>'
+        )
+        chat_lines.append(
+            f'Your {date_string} chat _"{penny_chat["title"]}"_ has follow-ups from {people_string}. '
+            f'*<http://{settings.PENNYU_DOMAIN}/chats/{penny_chat["id"]}|Read them here!>*'
```

Comment:
As with my comment above, could be replaced with FRONT_END_HOST

Reply:
Also, I wonder if a button would be better UX here. Do you have screen shots of what this looks like?

Reply:
I agree. Changed in 6985f40